week-10/10-tech-lab.adoc

```diff
@@ -0,0 +1,228 @@
+
+== Objectives
+
+In this lab, you will gain some hands on experience in vulnerability scanning and remediation.
+
+Objectives:
+
+* Understand a "vulnerability" and how it can be detected
+* Understand how a vulnerability may be fixed or remediated
+
+== Overview
+
+Working individually, you will run an "inside-out" vulnerability scan and a simple "outside-in" penetration scan on a virtual machine.  
+
+The testing will be performed by downloading a vulnerability scanner and running it on your server.
+
+You will research the results and recommend at least one fix that can be implemented.
+
+You will fix at least two vulnerabilities and participate in group and class discussion about how this relates to the broader topics of governance, risk, and compliance.
+
+
+=== Getting started
+
+Start up a Virtual Machine, login to your personal server, perform an update and install lynis.
+Lynis will not be installed in the directory you are currently in.  Use the 'find' command to determine where it was installed.
+Once you know where it got downloaded, run 'audit system' with the --quick option (don't wait for user input).  Lynis will generate hundreds
+of findings that will scroll off your display.  Redirect the lynis output to a file, e.g. lynis_out, so you can view it later.
+
+....
+cd ~
+sudo apt-get update
+sudo apt-get install lynis
+sudo find / -name lynis
+....
+
+Examine the locations to find the executable lynis*.  Once you know where it got downloaded, run 'audit system' with the --quick option (don't wait for user input).  
+Lynis will generate hundreds of findings that will scroll off your display.  Redirect the lynis output to a file, e.g. lynis_out, so you can view it later.
+
+....
+sudo <lynis* directory>/lynis --quick audit system > lynis_out
+....
+
+The lynis scan is run internally on the server, as a script or application that reviews the server's configuration in painstaking detail.   
+It will generate hundreds of findings.  You should pick one or two and research them and figure out if there is an easy fix.
+
+This scan is run internally on the server, as a script or application that reviews the 
+server's configuration in painstaking detail. Lots of output that looks like:
+
+....
+- Checking for old files in /tmp                            [ OK ]
+ - Checking /tmp sticky bit                                 [ OK ]
+ - Mount options of /                                       [ OK ]
+ - Checking Locate database                                 [ FOUND ]
+.
+.
+.
+
+ * Consider hardening SSH configuration [SSH-7408]
+    - Details  : AllowAgentForwarding (YES --> NO)
+      https://cisofy.com/controls/SSH-7408/
+
+  * Configure minimum password age in /etc/login.defs [AUTH-9286]
+      https://cisofy.com/controls/AUTH-9286/
+
+  * Configure maximum password age in /etc/login.defs [AUTH-9286]
+      https://cisofy.com/controls/AUTH-9286/
+....
+
+Let's fix the minium and maxium password age issue in login.defs.
```

Comment:
Did you validate that this was an issue on the current default Google VM? And the fix is still relevant?

Reply:
This issue is still relevant along with the fix.  Do you want me to add an image of the before and after result?